discoverer/src/main/java/eu/seaclouds/platform/discoverer/core/UniqueFileGen.java

```diff
@@ -32,9 +34,13 @@
 public class UniqueFileGen {
```

Comment:
do we really need this complicated class to generate a unique file. I think less code to maintain is always better.
If so, make sure you have a good test coverage for it